platform/admin/core/admin-core-configpolicy/src/main/java/org/codice/ddf/admin/config/policy/AdminConfigPolicy.java

```diff
@@ -0,0 +1,174 @@
+            KeyValueCollectionPermission matchAllCollection) {
+        return isPermittedMatchOne(subjectAllCollection, matchAllCollection);
+    }
+
+    @Override
+    public KeyValueCollectionPermission isPermittedMatchOne(
+            CollectionPermission subjectAllCollection,
+            KeyValueCollectionPermission matchOneCollection) {
+
+        if (matchOneCollection.getAction() == null || (!matchOneCollection.getAction()
+                .equals(VIEW_FEATURE_ACTION) && !matchOneCollection.getAction()
+                .equals(VIEW_SERVICE_ACTION))) {
+            return matchOneCollection;
+        }
+
+        KeyValueCollectionPermission newMatchCollection = new KeyValueCollectionPermission();
+
+        for (Permission permission : matchOneCollection.getPermissionList()) {
+
+            if (!(permission instanceof KeyValuePermission)) {
+                continue;
+            }
+
+            String matchPermissionName = ((KeyValuePermission) permission).getKey();
+
+            Map<String, List<KeyValueCollectionPermission>> policyPermissions;
+
+            if (matchPermissionName.equals(FEATURE_NAME)) {
+                policyPermissions = featurePolicyPermissions;
+            } else if (matchPermissionName.equals(SERVICE_PID)) {
+                policyPermissions = servicePolicyPermissions;
+            } else {
+                //Not a known policy extension key
+                continue;
+            }
+
+            boolean isPermitted = true;
+
+            //Typically only one feature or service is desired to be permitted at a time but there is support for multiple feature
+            //If there are multiple features in the permission and one is not authorized, the user is not authorized to see any of the features in the group
+            for (String matchPermissionValue : ((KeyValuePermission) permission).getValues()) {
+                List<KeyValueCollectionPermission> matchOneAttributes = policyPermissions.get(
+                        matchPermissionValue);
+
+                //Not a known restricted attribute, the feature or service is white listed
+                if (matchOneAttributes == null) {
+                    continue;
+                }
+
+                boolean matchedOneAttribute = false;
+
+                for (KeyValueCollectionPermission attributePermissions : matchOneAttributes) {
+                    if (subjectAllCollection.implies(attributePermissions)) {
+                        matchedOneAttribute = true;
+                        break;
+                    }
+                }
+
+                if (!matchedOneAttribute) {
+                    isPermitted = false;
+                    break;
+                }
+            }
+
+            if (!isPermitted) {
+                newMatchCollection.addAll(Collections.singleton(permission));
+            }
+
+        }
+        return newMatchCollection;
+    }
+
+    public void setFeaturePolicies(List<String> featurePolicies) {
```

Comment:
It looks like the following methods are only used by the test.  If so please remove this from the policy and move them to the test class instead.